python/lsst/ts/ess/csc/data_client/young_32400_weather_station_data_client.py

```diff
@@ -0,0 +1,713 @@
+    scale : `float`
+        Scale to apply to raw values: scaled = raw * scale + offset
+    offset : `float`:
+        Offset to apply to raw values: scaled = raw * scale + offset
+    """
+
+    def __init__(self, num_samples: int, scale: float, offset: float) -> None:
+        if num_samples < 1:
+            raise ValueError(f"{num_samples=} must be >= 1")
+        self.num_samples = num_samples
+        self.scale = scale
+        self.offset = offset
+        self.values: list[float] = []
+
+    def add_sample(self, value: float) -> float | None:
+        """Add a raw value. Return the scaled median, if enough samples
+        have been accumulated, else None.
+
+        Parameters
+        ----------
+        raw : `float`
+            Raw value to scale and to accumulate.
+
+        Returns
+        -------
+        median : `float` | `None`
+            The median scaled value, if enough samples have been accumulated
+            (in which case the accumulator is reset) else None.
+        """
+        self.values.append(value)
+        if len(self.values) >= self.num_samples:
+            median = np.median(self.values)
+            self.values = []
+            return median * self.scale + self.offset
+        return None
+
+
+def float_to_intstr(value: float, max_int: int) -> str:
+    """Return a float value converted to an string representation
+    of an integer with 4 chars and leading zeros.
+
+    Parameters
+    ----------
+    value : `float`
+        The value to convert.
+    max_int : `int`
+        The maximum integer value. Must be > 0 and <= 9999.
+
+    Raises
+    ------
+    ValueError
+        If max_int <= 0 or > 9999.
+    """
+    if not 0 < max_int <= 9999:
+        raise ValueError(f"{max_int=} not >0 and <= 9999")
+    int_val = int(round(value))
+    truncated_int_val = max(0, min(int_val, max_int))
+    return f"{truncated_int_val:04d}"
+
+
+def scaled_from_raw(raw_value: float, scale: float, offset: float) -> float:
+    """Convert raw data to scaled: return raw * scale + offset."""
+    return raw_value * scale + offset
+
+
+class Young32400WeatherStationDataClient(common.BaseDataClient):
+    """Get environmental data from Young 32400 weather station
+    serial interface.
+
+    The interface is assumed to be connected to a serial-to-ethernet adapter.
+
+    Parameters
+    ----------
+    config : types.SimpleNamespace
+        The configuration, after validation by the schema returned
+        by `get_config_schema` and conversion to a types.SimpleNamespace.
+    topics : `salobj.Controller` or `types.SimpleNamespace`
+        The telemetry topics this data client can write,
+        as a struct with attributes such as ``tel_temperature``.
+    log : `logging.Logger`
+        Logger.
+    simulation_mode : `int`, optional
+        Simulation mode; 0 for normal operation.
+
+    Notes
+    -----
+    This code assumes the 32400 is configured to provide ASCII or
```

Comment:
This documentation belongs in ts_ess_controller together with all other sensor documentation.

Reply:
OK. Though I am uncomfortable having the docs live so far from the code.